vars/pythonSetupPyPipeline.groovy

```diff
@@ -49,96 +61,212 @@ def call(Map pipelineParams) {
 
     stages {
       stage("Bump version Release") {
+       agent {
+          dockerfile {
+            filename pipelineParams.dockerFilename
+            additionalBuildArgs pipelineParams.dockerBuildArgs
+            args pipelineParams.dockerRunArgs
+            reuseNode true
+          }
+        }
         when {
           beforeAgent true
           allOf {
             branch "master"
-            expression {
-              params.doRelease
-            }
+            expression { params.doRelease }
           }
         }
         steps {
           sh "bumpversion release"
         }
-      }
+      } // Bump version Release
 
       stage("Build") {
-        steps {
-          sh "python setup.py build"
-        }
-      }
-
-      stage("Execute Pylint") {
-        steps {
-          // Necessary to avoid issues with creating pylint files that track the delta of warnings.
-          withEnv(["HOME=$WORKSPACE"]) {
-            // setuptools-lint does not make a good job in installing all its dependencies.
-            sh "python setup.py install --user"
-            sh "python setup.py lint --lint-output-format parseable"
+        agent {
+          dockerfile {
+            filename pipelineParams.dockerFilename
+            additionalBuildArgs pipelineParams.dockerBuildArgs
+            args pipelineParams.dockerRunArgs
+            reuseNode true
           }
         }
-        post {
-          always {
-            recordIssues enabledForFailure: true, tool: pyLint(), healthy: 0
-          }
-        }
-      }
+        stages {
+          stage("Build") {
+            parallel {
+              stage("Build module") {
+                steps {
+                  sh "python setup.py build"
 
-      stage("Execute Tests") {
-        steps {
-          sh "python setup.py test --addopts '--cov-report xml:build/coverage.xml --cov-report term --cov-branch --junitxml=build/test_results.xml'"
+                  script {
```

Comment:
get module name and version (or set) could be a separate stage

Reply:
I don't think that makes sense to have a separate stage for these two things. I have moved it to the Package stage now as it is the closest moment after that the two variables will be used.